src/realm/alloc_slab.cpp

```diff
@@ -360,15 +361,23 @@ MemRef SlabAlloc::do_alloc(const size_t size)
     // Add to list of slabs
     Slab slab;
     slab.addr = mem.get();
-    slab.ref_end = ref + new_size;
+    slab.ref_end = ref;
+    if (REALM_UNLIKELY(int_add_with_overflow_detect(slab.ref_end, new_size))) {
```

Comment:
Wouldn't it be better to have the check before calling operator new() ? If we throw here, the newly allocated block is leaked.